core/src/main/java/org/apache/calcite/avatica/ha/ShuffledRoundRobinLBStrategy.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.avatica.ha;
+
+import org.apache.calcite.avatica.ConnectionConfig;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+/**
+ * Shuffled Round Robin strategy for client side load balancing.
+ * It starts with some random position in a list of URLs, and then returns subsequent URL
+ * in a RoundRobin manner.
+ * It's implemented it as a singleton so that we can maintain state
+ * i.e. which URL was last used from the list of URLs specified.
+ */
+public class ShuffledRoundRobinLBStrategy implements LBStrategy {
+  private static final Logger LOG = LoggerFactory.getLogger(ShuffledRoundRobinLBStrategy.class);
+
+  public static final ShuffledRoundRobinLBStrategy INSTANCE = new ShuffledRoundRobinLBStrategy();
+  private ShuffledRoundRobinLBStrategy() { }
+  public static final String URL_SEPERATOR_CHAR = ",";
+
+  Map<String, Integer> configToIndexServedMap = new HashMap<>();
+  Map<String, String[]> configToUrlListMap = new HashMap<>();
+
+  @Override
+  public String getLbURL(ConnectionConfig config) {
+    String key = getKey(config);
+    String lbURLs = config.getLbURLs();
+    if (!configToIndexServedMap.containsKey(key)) {
+      initialiseCaches(key, lbURLs);
+    }
+    String[] urls = configToUrlListMap.get(key);
+    int urlIndex = configToIndexServedMap.get(key);
+    int nextURlIdx = urlIndex < urls.length - 1 ? (urlIndex + 1) : 0;
+    configToIndexServedMap.put(key, nextURlIdx);
+    String url = urls[urlIndex];
+    LOG.info("Selected URL:{}", url);
+    return url;
+  }
+
+  private void initialiseCaches(String key, String lbURLs) {
+    String[] urls = lbURLs.split(URL_SEPERATOR_CHAR);
+    List<String> list = Arrays.asList(urls);
+    Collections.shuffle(list);
+    list.toArray(urls);
```

Comment:
this doesn't do anything.

Reply:
@stoty 
It works as follows
1. Convert from array to list using Arrays.asList()
2. Collection.shuffle shuffles the list passed as an argument ( void return)
3. List.toArray( array ) - Put the shuffled elements in original array. [Check this](https://devdocs.io/openjdk~8/java/util/list). We  can ignored the return because we know that passed array will have shuffled elements.
Please let me know if you see any issue in above behavior or other suggestion.

Reply:
You are correct.
However, relying on that side effect hurts readability.
`list = Collections.shuffle(list);`
is obviously correct, and does the same thing.